### PR TITLE
api: display error type on screen if create/restore fails

### PR DESCRIPTION
**Pull request by benma**
If there was a securechip or memory error, creating or restoring a
wallet would show and return a generic looking error. If e.g. the
securechip is the problem, we want to see that, so the user can better
report the problem.

We might also return a specific error code to the host, but for now
displaying the error will do (same as do when unlocking a device fails).

### CODE REVIEW

File: src/keystore.c [hunk 1]
```diff
@@ -234,58 +234,59 @@ static bool _verify_seed(
...
     const char* password)
 {
     if (memory_is_initialized()) {
-        return false;
+        return KEYSTORE_ERR_MEMORY;
     }
     keystore_lock();
     if (!_validate_seed_length(seed_length)) {
-        return false;
+        return KEYSTORE_ERR_MEMORY;
```
Comment (Beerosagos) [inline src/keystore.c:247]: Shouldn't this be `KEYSTORE_ERR_SEEDSIZE`?

File: src/keystore.c [hunk 2]
```diff
@@ -234,58 +234,59 @@ static bool _verify_seed(
...
     }
     if (encrypted_seed_len > 255) { // sanity check, can't happen
         Abort("keystore_encrypt_and_store_seed");
     }
     if (!memory_set_encrypted_seed_and_hmac(encrypted_seed, encrypted_seed_len)) {
-        return false;
+        return KEYSTORE_ERR_MEMORY;
     }
     if (!_verify_seed(password, seed, seed_length)) {
         if (!memory_reset_hww()) {
```
Comment (Beerosagos) [inline src/keystore.c:275]: The `if` statement can be removed?

File: src/rust/bitbox02/src/keystore.rs [hunk 3]
```diff
@@ -39,12 +39,28 @@ pub enum Error {
...
     Salt,
     Hash,
     SeedSize,
-    Unknown,
+    Encrypt,
+}
+
+impl core::convert::From<keystore_error_t> for Error {
+    fn from(error: keystore_error_t) -> Self {
+        match error {
```
Comment (Beerosagos) [inline src/rust/bitbox02/src/keystore.rs:52]: `KEYSTORE_ERR_INCORRECT_PASSWORD` missing?

File: src/keystore.c (hunk 1 above)
Comment (benma) [reply src/keystore.c]: Good catch, fixed!

**Pull request merged**